libfs/fs.go

```diff
@@ -714,8 +714,8 @@ func (fs *FS) Chtimes(name string, atime time.Time, mtime time.Time) (
 	return fs.config.KBFSOps().SetMtime(fs.ctx, n, &mtime)
 }
 
-// Chroot implements the billy.Filesystem interface for FS.
-func (fs *FS) Chroot(p string) (newFS billy.Filesystem, err error) {
+// Chrute returns a *FS whose root is p.
```

Comment:
Cute name, but we shouldn't use cute names for exports.
Here are some alternatives:
1. You can use `Chroot` and allow the caller to type assert `newFS.(*libfs.FS)`.
2. You can edit `Chroot` to return the concrete type instead.
3. You can name `Chrute` => `ChrootFS`.

Reply:
I don't think 1 and 2 would work since it needs to satisfy the billy interface. Will go with rename path.

Reply:
I don't understand... if `Chroot` can passthrough to `ChrootLibFS`, then a type assert should work.

Reply:
The `Chroot` in `billy.FileSystem` expects a return type of `billy.FileSystem`. This is the same issue when migrating to stdlib's `context`. See: https://play.golang.org/p/XeWVEJCyM9p

Reply:
I know, the recommendation was to leave `Chroot` alone, and then assert its return type in the `libpages.CacheableFS.Use()` caller.

Reply:
Oh I see; I misunderstood.

Reply:
Not going to do that? It seems better--no abstraction leakage in the API.

Reply:
Oh I thought you were OK with renaming too. I just didn't want to have that type assertion; seemed unnecessary and it casts an assumption about the underlying type that's not guaranteed from method definition, which means if we change it later it'd break in `libpages`.

Reply:
Yeah the tradeoffs seem to be:
* `ChrootLibFS`:
  * (+) Validates typing at compile time
  * (-) Leaks the `*FS` concrete type
  * (-) `FS` is no longer just an implementation of `billy.Filesystem`
* `Chroot` + type assert:
  * (-) Doesn't validate at compile time, can fail at runtime
  * (+) Preserves the `billy.Filesystem` abstraction
  * (+) Doesn't create a new API to support solely for a custom downstream use case

I guess it doesn't really matter.

Reply:
Nice summary! Just one thing:

> (-) FS is no longer just an implementation of billy.Filesystem

Unfortunately we already have `ToHTTPFileSystem` on `*libfs.FS` (also for the purpose of typing), so this one's already broken.